streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -172,13 +173,15 @@ public ProcessorNode build() {
         private final Pattern pattern;
         private final Deserializer<?> keyDeserializer;
         private final Deserializer<?> valDeserializer;
+        private final TimestampExtractor timestampExtractor;
 
-        private SourceNodeFactory(String name, String[] topics, Pattern pattern, Deserializer<?> keyDeserializer, Deserializer<?> valDeserializer) {
+        private SourceNodeFactory(String name, String[] topics, Pattern pattern, TimestampExtractor timestampExtractor, Deserializer<?> keyDeserializer, Deserializer<?> valDeserializer) {
```

Comment:
I know you've only added the one param here, but seeing as you are changing it can you make all the params `final`?